conanfile.py

```diff
@@ -17,38 +18,75 @@ class WinflexbisonConan(ConanFile):
-        extracted_dir = self.name + "-" + self.version
+        name = "winflexbison"
+        url = "https://github.com/lexxmark/winflexbison/archive/v{}.tar.gz".format(self.version)
+        sha256 = "2ab4c895f9baf03dfdfbb2dc4abe60e87bf46efe12ed1218c38fd7761f0f58fc"
+        filename = "{}-{}.tar.gz".format(name, self.version)
+
+        dlfilepath = os.path.join(tempfile.gettempdir(), filename)
+        if os.path.exists(dlfilepath) and not tools.get_env("WINFLEXBISON_FORCE_DOWNLOAD", False):
+            self.output.info("Skipping download. Using cached {}".format(dlfilepath))
+        else:
+            self.output.info("Downloading {} from {}".format(name, url))
+            tools.download(url, dlfilepath)
+        tools.check_sha256(dlfilepath, sha256)
+        tools.untargz(dlfilepath)
+
+        extracted_dir = "{}-{}".format(name, self.version)
         os.rename(extracted_dir, self._source_subfolder)
+
         # Generate license from header of a source file
-        with open("%s/%s/bison/data/glr.cc" % (self.source_folder, self._source_subfolder)) as f:
+        with open(os.path.join(self.source_folder, self._source_subfolder, "bison", "data", "skeletons", "glr.cc"), ) as f:
             content_lines = f.readlines()
         license_content = []
         for i in range(2, 16):
             license_content.append(content_lines[i][2:-1])
-        tools.save("%s/%s/LICENSE" % (self.source_folder, self._source_subfolder), "\n".join(license_content))
+        tools.save(os.path.join(self.source_folder, self._source_subfolder, "COPYING"), "\n".join(license_content))
+
+    @property
+    def _fake_package(self):
+        return os.path.join(self.build_folder, "package")
 
     def build(self):
         cmake = CMake(self)
+        cmake.definitions["CMAKE_INSTALL_PREFIX"] = self._fake_package
         cmake.configure()
-        cmake.build()
+        cmake.build(args=["--config", str(self.settings.build_type)])
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="data/*", dst="bin", src="{}/bison".format(self._source_subfolder), keep_path=True)
-        actual_build_path = "{}/bin/{}".format(self._source_subfolder, self.settings.build_type)
-        self.copy(pattern="*.exe", dst="bin", src=actual_build_path, keep_path=False)
-        self.copy(pattern="*.h", dst="include", src=actual_build_path, keep_path=False)
+        cmake = CMake(self)
+        with tools.chdir(self.build_folder):
+            try:
+                os.mkdir(self._fake_package)
+            except IOError:
+                pass
+            cmake.install(args=["--config", str(self.settings.build_type)])
+
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+
+        self.copy(pattern="*.exe", src=self._fake_package, dst="bin",keep_path=False)
```

Comment:
What is `self._face_package` about?

Again, please stay with the templates as close as possible. We have way too many recipes as custom logic would be easily maintainable.

Reply:
Instead of copying sources out of the build tree, I lest the package install itself.
Because the install procedure does use `bin`, `include`, `share` subdirectories,
I install to a `fake package`, from which the files are copied from in `package()`

Reply:
I don't understand, maybe i'm missing something.  If the project has a proper cmake install method defined, we should be able to just call `cmake.install()` , and the files should be copied directly to `package` sub-directories by default.  I'm trying to use my imagination, but can't see any advantage to having a `fake package` directory or function or anything. Are we eliminating a copy of some files, or working around some conflicting filenames or something?

Reply:
I use the `fake_package`  directory because the default `install` step does not copy the built files to the `bin` subdirectories.

Reply:
The [install rules in the cmake file install the files in the root package folder](
https://github.com/lexxmark/winflexbison/blob/a72f3d6b5102b65f064a5054ba634d3d62e94f41/CMakeLists.txt#L37-L47).
You can either `tools.replace_in_file` all these lines, or simply copy these files manually after it installed itself. I choose the latter because it is easier.